### PR TITLE
IEP-1153: Configure GitHub and PyWheels mirrors

### PR DESCRIPTION
**Pull request by alirana01**
## Description

Added a setting to configure the mirrors for tools installation in the Espressif Preference page to help users configure the installation of the tools via a different mirrors

Fixes # ([IEP-1153](https://jira.espressif.com:8443/browse/IEP-1153))

## Type of change

Please delete options that are not relevant.
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## How has this been tested?

Configure the mirror for the Github and python pip wheels to the cn server and install tools make sure to clear up the pip caches on your system and the espressif home directory. Install tools observe that the console should be downloading the tools from the defined mirror

image

**Test Configuration**:
* ESP-IDF Version: any
* OS (Windows,Linux and macOS): all

## Checklist
- [x] PR Self Reviewed
- [x] Applied Code formatting
- [x] Added Documentation
- [ ] Verified on all platforms - Windows,Linux and macOS

## Summary by CodeRabbit

- **New Features**
	- Introduced new preference settings for configuring GitHub assets and Python package installation sources.
	- Added tools installation settings UI elements for GitHub assets and Python wheels.

- **Enhancements**
	- Improved tools installation settings within the preferences page for better user customization.
	- Updated UI layout and preference handling for a more user-friendly experience.

- **Documentation**
	- Added information in the FAQ about setting mirrors for Git and Pip Py Wheels to resolve timeout errors during tool installation.
	- Provided instructions for adjusting mirrors based on region.

### CODE REVIEW

**Review by alirana01 — COMMENTED** [review]
Self Review

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/preferences/EspresssifPreferencesPage.java [hunk 1]
```diff
@@ -69,9 +72,44 @@ protected Control createContents(Composite parent)
...
+
+		Label githubAssetsLabel = new Label(toolsInstallationGroup, SWT.NONE);
+		githubAssetsLabel.setText("Mirror for GIT (IDF_GITHUB_ASSETS)");
+		gitAssetsText = new Text(toolsInstallationGroup, SWT.SINGLE | SWT.BORDER);
+		
+		Label pythonWheelsLabel = new Label(toolsInstallationGroup, SWT.NONE);
+		pythonWheelsLabel.setText("Mirror for python wheels (PIP_EXTRA_INDEX_URL)");
+		pythonWheelText = new Text(toolsInstallationGroup, SWT.SINGLE | SWT.NONE);
+		
+		GridData gitTextGridData = new GridData(SWT.FILL, SWT.CENTER, true, false);
```
Comment (sigmaaa) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/preferences/EspresssifPreferencesPage.java:94]: We could externalize these UI labels. Also, seems like the formatting is missing here.

**Review by sigmaaa — APPROVED** [review]
LGTM

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/preferences/EspresssifPreferencesPage.java (hunk 1 above)
Comment (alirana01) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/preferences/EspresssifPreferencesPage.java]: Done

**Review by kolipakakondal — APPROVED** [review]
LGTM.

I think we need to clarify in the documentation that if these environment variables are configured at the system level or in the CDT Build environment, they will not be considered. One must configure alternative servers only through this preference option.

**Comment by kolipakakondal** [comment]
@alirana01 Please rebase the changes with the master to resolve the conflcits so that @AndriiFilippov could verify further.

**Comment by AndriiFilippov** [comment]
@alirana01 hi!

Tested under:

OS - Windows 10
ESP-IDF: v5.1.2

during tool installation:
image

**Comment by AndriiFilippov** [comment]
@alirana01 hi ! 

Tested under: 
OS - Windows 10 
ESP-IDF: v5.1.2 


able to install tools / pip using cn mirrors 🟢 
able to build / flash / debug / monitor 🟢 

LGTM 👍

**Pull request merged**